File: sandbox/exercism/ruby/2024/raindrops.rb
# https://exercism.org/tracks/ruby/exercises/raindrops

class Raindrops
  SOUNDS = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong',
  }.freeze

  def self.convert(number)
    sounds = SOUNDS.map { |n, sound| sound if (number % n).zero? }
    sounds.any? ? sounds.join : number.to_s
  end
end

File: sandbox/exercism/go/2025/raindrops.go
// https://exercism.org/tracks/go/exercises/raindrops

package raindrops

import("strconv")

func Convert(number int) string {
	sound := ""

	if number % 3 == 0 { sound += "Pling" }
	if number % 5 == 0 { sound += "Plang" }
	if number % 7 == 0 { sound += "Plong" }

	if len(sound) > 0 { return sound }

	return strconv.Itoa(number)
}
